static/app/components/guidedSteps/guidedSteps.tsx

```diff
@@ -60,7 +63,26 @@ function useGuidedStepsContentValue({
 }: Pick<GuidedStepsProps, 'onStepChange'>): GuidedStepsContextState {
   const registeredStepsRef = useRef<RegisteredSteps>({});
   const [totalSteps, setTotalSteps] = useState<number>(0);
-  const [currentStep, setCurrentStep] = useState<number>(1);
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const currentStep = decodeInteger(location.query.guidedStep);
+
+  const setCurrentStep = useCallback(
+    (step: number) => {
+      if (step === currentStep) {
+        return;
+      }
+      navigate({
+        ...location,
+        query: {
+          ...location.query,
+          guidedStep: step,
+        },
+      });
+    },
+    [navigate, currentStep, location]
```

Comment:
The location is memorized, the currentStep is a `number`, and navigate uses `useCallback`